vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -172,7 +176,58 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         this.context.acquisitionContext = context;
     }
 
-    private async uninstallRuntime(version: string) {
+    /**
+     *
+     * @param installedVersions
+     * @param version
+     *
+     * @remarks Before, installed versions used their version as the 'install key' in the promises and folder structure.
+     * We changed this install key to include architecture so different architectures could be installed side-by-side.
+     * This means any installs that were made before version 1.7.4 will not have the architecture in their install key.
+     * They should be removed. This is what makes an install 'legacy'.
+     *
+     * This function only removes the legacy install with the same version as 'version'.
+     * That's because removing other legacy installs may cause a breaking change.
+     * Assuming the install succeeds, this will not break as the legacy install of 'version' will be replaced by a non-legacy one upon completion.
+     *
+     * Many (if not most) legacy installs will actually hold the same content as the newly installed runtime/sdk.
+     * But since we don't want to be in the business of detecting their architecture, we chose this option as opposed to renaming and install key and folder
+     * ... for the legacy install.
+     *
+     * Note : only local installs were ever 'legacy.'
+     */
+    private async removeAnyLegacyInstalls(installedVersions : string[], version : string)
+    {
+        const legacyInstalls = this.existingLegacyInstalls(installedVersions);
+        for(const legacyInstall of legacyInstalls)
+        {
+            if(legacyInstall.includes(version))
+            {
+                await this.uninstallRuntimeOrSDK(legacyInstall);
+            }
+        }
+    }
+
+    /**
+     *
+     * @param allInstalls all of the existing installs.
+     * @returns All existing installs made by the extension that dont include a - for the architecture.
+     */
+    private existingLegacyInstalls(allInstalls : string[]) : string[]
+    {
+        let legacyInstalls : string[] = [];
+        for(const install of allInstalls)
+        {
+            if(!install.includes('-'))
```

Comment:
Oh, that's a really good point. They definitely do. We can't use `/` because the install key becomes part of the path, similarly `:` is an invalid path character. So instead I think I'll use `~` which we don't use anywhere.